### PR TITLE
fix(Resources/AnomalyDetection/): fixes

### PR DESCRIPTION
**Pull request by Noha-ElAbrouki**
adding modal for closing confirmation : 
Closing the modal (click on cross or outside). If the user has confirmed his choice or made no changes, the modal closes without warning. Otherwise, popin “you have unsaved changed, do you want to proceed? cancel, confirm” (wording TBD)

**video:**

https://user-images.githubusercontent.com/97687698/194581233-9c296fcf-a1cc-43fc-a13a-088b1118ea27.mp4

**screens:**
image

image

### CODE REVIEW

**Comment by Thebarda** [comment]
There is a component in our design system for that ^^.
UnsavedChangesDialog or something like that

**Comment by bdauria** [comment]
@Noha-ElAbrouki can we try to have a rounded badge with the same size in all cases for the slider indicator?

**Comment by Noha-ElAbrouki** [comment]
> UnsavedChangesDialog

this component contains required props which doesn't fit my needs and it has text msg defined , i think it's dedicated for forms. the component used here is also from centreon-frontend (Dialog)

File: www/front_src/src/Resources/Graph/Performance/AnomalyDetection/AnomalyDetectionSlider.tsx [hunk 1]
```diff
@@ -290,20 +296,26 @@ const AnomalyDetectionSlider = ({
...
-          label={t(labelUseDefaultValue)}
-        />
+        <Button
+          className={classes.buttonSetDefault}
+          disabled={isDefaultValue}
+          size="small"
+          variant="text"
+          onClick={setToDefaultValue}
+        >
+          <ResetIcon color={isDefaultValue ? 'disabled' : 'primary'} />
```
Comment (bdauria) [inline www/front_src/src/Resources/Graph/Performance/AnomalyDetection/AnomalyDetectionSlider.tsx]: Should be is in start adornment

File: www/front_src/src/Resources/Graph/Performance/AnomalyDetection/AnomalyDetectionSlider.tsx [hunk 2]
```diff
@@ -40,6 +36,11 @@ const useStyles = makeStyles((theme) => ({
     marginBottom: theme.spacing(2),
     marginTop: theme.spacing(5),
   },
+  buttonSetDefault: {
+    color: theme.palette.text.primary,
+    justifyContent: 'flex-start',
+    textTransform: 'none',
+  },
```
Comment (bdauria) [inline www/front_src/src/Resources/Graph/Performance/AnomalyDetection/AnomalyDetectionSlider.tsx]: Use material default instead, with a non contained variant

File: lang/pt_BR.UTF-8/LC_MESSAGES/messages.po [hunk 3]
```diff
@@ -15816,3 +15816,9 @@ msgstr "Editar dados de detecção de anomalias"
 
 #  msgid "Host Groups (simplified)"
 #  msgstr ""
+
+msgid "you have unsaved changed, do you want to proceed?"
+msgstr "você não salvou as alterações, deseja continuar?"
+
+#  msgid "Confirm"
+#  msgstr "confirme"
```
Comment (Thebarda) [inline lang/pt_BR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion l'
#  msgstr "Confirme"
```

File: lang/pt_PT.UTF-8/LC_MESSAGES/messages.po [hunk 4]
```diff
@@ -15804,3 +15804,9 @@ msgstr "Editar dados de detecção de anomalias"
 
 #  msgid "Host Groups (simplified)"
 #  msgstr ""
+
+msgid "you have unsaved changed, do you want to proceed?"
+msgstr "você não salvou as alterações, deseja continuar?"
+
+#  msgid "Confirm"
+#  msgstr "confirme"
```
Comment (Thebarda) [inline lang/pt_PT.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
#  msgstr "Confirme"
```

File: www/front_src/src/Resources/Graph/Performance/AnomalyDetection/AnomalyDetectionSlider.tsx (hunk 1 above)
Comment (Thebarda) [reply www/front_src/src/Resources/Graph/Performance/AnomalyDetection/AnomalyDetectionSlider.tsx]: The prop is starIcon

File: www/front_src/src/Resources/Graph/Performance/AnomalyDetection/AnomalyDetectionSlider.tsx (hunk 2 above)
Comment (Noha-ElAbrouki) [reply www/front_src/src/Resources/Graph/Performance/AnomalyDetection/AnomalyDetectionSlider.tsx]: i used iConButton instead 🤔

File: www/front_src/src/Resources/Graph/Performance/AnomalyDetection/AnomalyDetectionSlider.tsx (hunk 1 above)
Comment (Noha-ElAbrouki) [reply www/front_src/src/Resources/Graph/Performance/AnomalyDetection/AnomalyDetectionSlider.tsx]: i used iConButton instead 🤔

File: www/front_src/src/Resources/Graph/Performance/AnomalyDetection/AnomalyDetectionSlider.tsx [hunk 5]
```diff
@@ -290,19 +294,24 @@ const AnomalyDetectionSlider = ({
...
-          label={t(labelUseDefaultValue)}
-        />
+        <IconButton
+          disabled={isDefaultValue}
+          size="small"
+          onClick={setToDefaultValue}
+        >
+          <ResetIcon />
+          <Typography>{t(labelUseDefaultValue)}</Typography>
+        </IconButton>
```
Comment (Thebarda) [inline www/front_src/src/Resources/Graph/Performance/AnomalyDetection/AnomalyDetectionSlider.tsx]: First use the Button component then use the startIcon prop: https://mui.com/material-ui/api/button/

File: lang/es_ES.UTF-8/LC_MESSAGES/messages.po [hunk 6]
```diff
@@ -15355,3 +15355,9 @@ msgstr "Editar datos de detección de anomalías"
 
 #  msgid "Host Groups (simplified)"
 #  msgstr ""
+
+msgid "you have unsaved changed, do you want to proceed?"
```
Comment (smau-centreon) [inline lang/es_ES.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgid "You have unsaved changed, do you want to proceed?"
```

File: lang/es_ES.UTF-8/LC_MESSAGES/messages.po [hunk 7]
```diff
@@ -15355,3 +15355,9 @@ msgstr "Editar datos de detección de anomalías"
...
+msgstr "no ha guardado los cambios, ¿quiere continuar?"
```
Comment (smau-centreon) [inline lang/es_ES.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgstr "No ha guardado los cambios, ¿quiere continuar?"
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 8]
```diff
@@ -17184,3 +17184,9 @@ msgstr "Groupes d'hôtes (simplifiés)"
 
 msgid "Services (simplified)"
 msgstr "Services (simplifiés)"
+
+msgid "you have unsaved changed, do you want to proceed?"
```
Comment (smau-centreon) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgid "You have unsaved changed, do you want to proceed?"
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 9]
```diff
@@ -17184,3 +17184,9 @@ msgstr "Groupes d'hôtes (simplifiés)"
...
+msgstr "vous avez des modifications non enregistrées, voulez-vous continuer?"
```
Comment (smau-centreon) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgstr "Vous avez des modifications non enregistrées, voulez-vous continuer?"
```

File: lang/pt_BR.UTF-8/LC_MESSAGES/messages.po [hunk 10]
```diff
@@ -15816,3 +15816,9 @@ msgstr "Editar dados de detecção de anomalias"
 
 #  msgid "Host Groups (simplified)"
 #  msgstr ""
+
+msgid "you have unsaved changed, do you want to proceed?"
```
Comment (smau-centreon) [inline lang/pt_BR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgid "You have unsaved changed, do you want to proceed?"
```

File: lang/pt_BR.UTF-8/LC_MESSAGES/messages.po [hunk 11]
```diff
@@ -15816,3 +15816,9 @@ msgstr "Editar dados de detecção de anomalias"
...
+msgstr "você não salvou as alterações, deseja continuar?"
```
Comment (smau-centreon) [inline lang/pt_BR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgstr "Você não salvou as alterações, deseja continuar?"
```

File: lang/pt_PT.UTF-8/LC_MESSAGES/messages.po [hunk 12]
```diff
@@ -15804,3 +15804,9 @@ msgstr "Editar dados de detecção de anomalias"
 
 #  msgid "Host Groups (simplified)"
 #  msgstr ""
+
+msgid "you have unsaved changed, do you want to proceed?"
```
Comment (smau-centreon) [inline lang/pt_PT.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgid "You have unsaved changed, do you want to proceed?"
```

File: lang/pt_PT.UTF-8/LC_MESSAGES/messages.po [hunk 13]
```diff
@@ -15804,3 +15804,9 @@ msgstr "Editar dados de detecção de anomalias"
...
+msgstr "você não salvou as alterações, deseja continuar?"
```
Comment (smau-centreon) [inline lang/pt_PT.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgstr "Você não salvou as alterações, deseja continuar?"
```

**Review by smau-centreon — CHANGES_REQUESTED** [review]

**Review by bdauria — APPROVED** [review]

**Review by smau-centreon — APPROVED** [review]

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 14]
```diff
@@ -17184,3 +17184,9 @@ msgstr "Groupes d'hôtes (simplifiés)"
 
 msgid "Services (simplified)"
 msgstr "Services (simplifiés)"
+
+msgid "You have unsaved changed, do you want to proceed?"
```
Comment (cg-tw) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgid "You have unsaved changes, do you want to proceed?"
```

File: lang/pt_BR.UTF-8/LC_MESSAGES/messages.po [hunk 15]
```diff
@@ -15816,3 +15816,9 @@ msgstr "Editar dados de detecção de anomalias"
 
 #  msgid "Host Groups (simplified)"
 #  msgstr ""
+
+msgid "You have unsaved changed, do you want to proceed?"
```
Comment (cg-tw) [inline lang/pt_BR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgid "You have unsaved changes, do you want to proceed?"
```

File: lang/pt_PT.UTF-8/LC_MESSAGES/messages.po [hunk 16]
```diff
@@ -15804,3 +15804,9 @@ msgstr "Editar dados de detecção de anomalias"
 
 #  msgid "Host Groups (simplified)"
 #  msgstr ""
+
+msgid "You have unsaved changed, do you want to proceed?"
```
Comment (cg-tw) [inline lang/pt_PT.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgid "You have unsaved changes, do you want to proceed?"
```

File: www/front_src/src/Resources/translatedLabels.ts [hunk 17]
```diff
@@ -270,6 +270,9 @@ export const labelEditAnomalyDetectionConfirmation =
 export const labelMenageEnvelope = 'Manage envelope size';
 export const labelMenageEnvelopeSubTitle =
   'Changes to the envelope size will be applied immediately';
-export const labelUseDefaultValue = 'use default value';
+export const labelUseDefaultValue = 'Set to default value';
 export const labelPointsOutsideOfEnvelopeCount = 'points out of envelope';
 export const labelClose = 'Close';
+export const labelEditAnomalyDetectionClosing =
+  'you have unsaved changed, do you want to proceed?';
```
Comment (cg-tw) [inline www/front_src/src/Resources/translatedLabels.ts]:
```suggestion
  'You have unsaved changes, do you want to proceed?';
```

File: lang/es_ES.UTF-8/LC_MESSAGES/messages.po [hunk 18]
```diff
@@ -15355,3 +15355,9 @@ msgstr "Editar datos de detección de anomalías"
 
 #  msgid "Host Groups (simplified)"
 #  msgstr ""
+
+msgid "You have unsaved changed, do you want to proceed?"
```
Comment (cg-tw) [inline lang/es_ES.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgid "You have unsaved changes, do you want to proceed?"
```

**Review by cg-tw — CHANGES_REQUESTED** [review]

File: www/front_src/src/Resources/translatedLabels.ts [hunk 19]
```diff
@@ -270,6 +270,9 @@ export const labelEditAnomalyDetectionConfirmation =
 export const labelMenageEnvelope = 'Manage envelope size';
 export const labelMenageEnvelopeSubTitle =
   'Changes to the envelope size will be applied immediately';
-export const labelUseDefaultValue = 'use default value';
+export const labelUseDefaultValue = 'Set to default value';
```
Comment (Thebarda) [inline www/front_src/src/Resources/translatedLabels.ts]:
```suggestion
export const labelSetToDefaultValue = 'Set to default value';
```

File: www/front_src/src/Resources/translatedLabels.ts [hunk 20]
```diff
@@ -270,6 +270,9 @@ export const labelEditAnomalyDetectionConfirmation =
...
 export const labelPointsOutsideOfEnvelopeCount = 'points out of envelope';
```
Comment (cg-tw) [inline www/front_src/src/Resources/translatedLabels.ts]:
```suggestion
export const labelPointsOutsideOfEnvelopeCount = 'points outside of the envelope';
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 21]
```diff
@@ -17143,8 +17143,8 @@ msgstr "Les modifications apportées à la taille de l'enveloppe seront appliqu
 msgid "Manage envelope size"
 msgstr "Gérer la taille de l'enveloppe"
 
-msgid "use default value"
-msgstr "utiliser la valeur par défaut"
+msgid "Set to default value"
+msgstr "Définir sur la valeur par défaut"
```
Comment (cg-tw) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgstr "Rétablir la valeur par défaut"
```

File: www/front_src/src/Resources/translatedLabels.ts (hunk 19 above)
Comment (cg-tw) [inline www/front_src/src/Resources/translatedLabels.ts]:
```suggestion
export const labelUseDefaultValue = 'Reset to default value';
```

File: lang/es_ES.UTF-8/LC_MESSAGES/messages.po [hunk 22]
```diff
@@ -15317,8 +15317,8 @@ msgstr "Los cambios en el tamaño del sobre se aplicarán inmediatamente."
 msgid "Manage envelope size"
 msgstr "Administrar el tamaño del sobre"
 
-msgid "use default value"
-msgstr "usar el valor predeterminado"
+msgid "Set to default value"
```
Comment (cg-tw) [inline lang/es_ES.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgid "Reset to default value"
```

File: lang/es_ES.UTF-8/LC_MESSAGES/messages.po [hunk 23]
```diff
@@ -15317,8 +15317,8 @@ msgstr "Los cambios en el tamaño del sobre se aplicarán inmediatamente."
...
+msgstr "Establecer en el valor predeterminado"
 
 msgid "points out of envelope"
```
Comment (cg-tw) [inline lang/es_ES.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgid "points outside of the envelope"
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 24]
```diff
@@ -17143,8 +17143,8 @@ msgstr "Les modifications apportées à la taille de l'enveloppe seront appliqu
 msgid "Manage envelope size"
 msgstr "Gérer la taille de l'enveloppe"
 
-msgid "use default value"
-msgstr "utiliser la valeur par défaut"
+msgid "Set to default value"
```
Comment (cg-tw) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgid "Reset to default value"
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 25]
```diff
@@ -17143,8 +17143,8 @@ msgstr "Les modifications apportées à la taille de l'enveloppe seront appliqu
...
+msgstr "Définir sur la valeur par défaut"
 
 msgid "points out of envelope"
```
Comment (cg-tw) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgid "points outside of the envelope"
```

File: lang/pt_BR.UTF-8/LC_MESSAGES/messages.po [hunk 26]
```diff
@@ -15778,8 +15778,8 @@ msgstr "As alterações no tamanho do envelope serão aplicadas imediatamente"
 msgid "Manage envelope size"
 msgstr "Gerenciar o tamanho do envelope"
 
-msgid "use default value"
-msgstr "usar valor padrão"
+msgid "Set to default value"
```
Comment (cg-tw) [inline lang/pt_BR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgid "Reset to default value"
```

File: lang/pt_BR.UTF-8/LC_MESSAGES/messages.po [hunk 27]
```diff
@@ -15778,8 +15778,8 @@ msgstr "As alterações no tamanho do envelope serão aplicadas imediatamente"
...
+msgstr "Definir para o valor padrão"
 
 msgid "points out of envelope"
```
Comment (cg-tw) [inline lang/pt_BR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgid "points outside of the envelope"
```

File: lang/pt_PT.UTF-8/LC_MESSAGES/messages.po [hunk 28]
```diff
@@ -15766,8 +15766,8 @@ msgstr "As alterações no tamanho do envelope serão aplicadas imediatamente"
 msgid "Manage envelope size"
 msgstr "Gerenciar o tamanho do envelope"
 
-msgid "use default value"
-msgstr "usar valor padrão"
+msgid "Set to default value"
```
Comment (cg-tw) [inline lang/pt_PT.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgid "Reset to default value"
```

File: lang/pt_PT.UTF-8/LC_MESSAGES/messages.po [hunk 29]
```diff
@@ -15766,8 +15766,8 @@ msgstr "As alterações no tamanho do envelope serão aplicadas imediatamente"
...
+msgstr "Definir para o valor padrão"
 
 msgid "points out of envelope"
```
Comment (cg-tw) [inline lang/pt_PT.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgid "points outside of the envelope"
```

**Review by cg-tw — CHANGES_REQUESTED** [review]

**Review by Thebarda — APPROVED** [review]

**Review by cg-tw — APPROVED** [review]

**Pull request merged**